src/extension.c

```diff
@@ -1,12 +1,20 @@
 #include <postgres.h>
 #include <access/xact.h>
 #include <access/transam.h>
+#ifdef _MSC_VER
```

Comment:
I am not sure how I like these ifdef in regular src files as opposed in src/msvc. Or maybe it should be src/compat/extension.h which references src/compat/msvc/extension.h? Don't feel strongly here. @erimatnor thoughts?

Reply:
Not sure what @erimatnor thinks but that seems like an annoying indirection to me if I'm trying to browse and understand the code. This is more straightforward while not being too messy, at least in my opinion.

Reply:
The main problem with defining them in-place is I run into redefining the symbol when I add `PGDLLIMPORT`. This also gets into why I need the extra things that are copied from the original source files - I can include either or compat definitions or the original ones, but not both. I'm also not sure how your hack below would help since the issue is redefining `constraint_exclusion` or `CurrentExtensionObject`, not `PGDLLIMPORT`. Perhaps I am misunderstanding.

I can move this `ifdef` to the .h file, but the original inclusion was here in the .c file so I figured it made more sense to keep it there.

Reply:
So, `extension.c` need only two extern declarations from the PG `extension.h`. You can do those declarations in `extension.c` without including any extension-related headers (in fact, this would be the same as having the pre-processor prepend your own `msvc/extension.h` with these declarations). This would remove the need for `msvc/extension.h` and would obviate the need for `ifdefs` in this case. The other extern variables from other headers might be tricker, but can be solved with the preprocessor hack I gave below that basically makes all extern variables use the PGDLLIMPORT storage class.